src/shell/Shell.cpp

```diff
@@ -325,7 +227,7 @@ static Trap::TrapResult executeWASM(Store* store, const std::string& filename, c
         } else if (import->moduleName() == "wasi_snapshot_preview1") {
             // TODO wasi
             if (import->fieldName() == "proc_exit") {
-                auto ft = functionTypes[SpecTestFunctionTypes::I32];
+                auto ft = functionTypes[SpecTestFunctionTypes::I32R];
                 importValues.push_back(ImportedFunction::createImportedFunction(
                     store,
                     ft,
```

Comment:
Looks good :)
Would you please merge this `proc_exit` wasi function into `Wasi.cpp`?

Reply:
Should be done. I also changed `basic.wast` to `example.wast` to demonstrate how functions can be made. It could be removed in the future though.